packages/cli/internal/pkg/cli/util/timeout.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	"errors"
+	"time"
+)
+
+const (
+	deploymentTimeout = 30 * time.Minute
+)
+
+func DeployWithTimeout(timeoutFunction func() error) error {
+	// channel to mark when a deployment successfully completes
+	completionChannel := make(chan error)
+	go func() {
+		err := timeoutFunction()
+		completionChannel <- err
+	}()
+	select {
+	case err := <-completionChannel:
+		return err
+	case <-time.After(deploymentTimeout):
```

Comment:
We discussed querying Cloudformation to validate the current state of the stack as it could have deployed successfully. Can you please implement that feature since the deployment is async?

Reply:
Yup, that should be a pretty simple addition